jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/LocalBaseImageSteps.java

```diff
@@ -88,27 +89,67 @@ static boolean isGzipped(Path path) throws IOException {
     }
   }
 
-  private final ExecutorService executorService;
-  private final BuildConfiguration buildConfiguration;
-  private final Path tarPath;
-  private final ProgressEventDispatcher.Factory progressEventDispatcherFactory;
-  private final TempDirectoryProvider tempDirectoryProvider;
+  static Callable<LocalImage> retrieveDockerDaemonImageStep(
+      ExecutorService executorService,
+      BuildConfiguration buildConfiguration,
+      ProgressEventDispatcher.Factory progressEventDispatcherFactory,
+      DockerClient dockerClient) {
+    return () -> {
+      ImageReference imageReference = buildConfiguration.getBaseImageConfiguration().getImage();
+      try (TempDirectoryProvider tempDirectoryProvider = new TempDirectoryProvider();
+          ProgressEventDispatcher progressEventDispatcher =
+              progressEventDispatcherFactory.create("processing base image " + imageReference, 2)) {
```

Comment:
I think the outermost `try` can be merged with the next `try`.

And unrelated, but why the progress unit is 2? Doesn't seem like we dispatch an extra event manually to make it 2.

Reply:
We create two child producers (one for docker save, one for `cacheDockerImageTar()`)